modules/test/playwright/tests/client-extension-web/main/fdsDataProvider.spec.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {dataApiHelpersTest} from '../../../fixtures/dataApiHelpersTest';
+import {isolatedSiteTest} from '../../../fixtures/isolatedSiteTest';
+import {loginTest} from '../../../fixtures/loginTest';
+import {pagesAdminPagesTest} from '../../../fixtures/pagesAdminPagesTest';
+import {siteSettingsPagesTest} from '../../../fixtures/siteSettingsPagesTest';
+import {usersAndOrganizationsPagesTest} from '../../../fixtures/usersAndOrganizationsPagesTest';
+import {getRandomInt} from '../../../utils/getRandomInt';
+import getRandomString from '../../../utils/getRandomString';
+import {clientExtensionsPageTest} from './fixtures/clientExtensionsPageTest';
+import {editJSClientExtensionsPageTest} from './fixtures/editJSClientExtensionsPageTest';
+import {WaitAction} from './pages/EditClientExtensionsPage';
+
+const test = mergeTests(
+	clientExtensionsPageTest,
+	dataApiHelpersTest,
+	editJSClientExtensionsPageTest,
+	isolatedSiteTest,
+	loginTest(),
+	pagesAdminPagesTest,
+	siteSettingsPagesTest,
+	usersAndOrganizationsPagesTest
+);
+
+test('Check users without permission cannot fetch client extension data', async ({
+	clientExtensionsPage,
+	editJSClientExtensionsPage,
+	page,
+	request,
+	usersAndOrganizationsPage,
+}) => {
+
+	// create client extension
+
+	await editJSClientExtensionsPage.goto();
+
+	const clientExtensionName = getRandomString();
+
+	await editJSClientExtensionsPage.nameInput.fill(clientExtensionName);
+
+	await editJSClientExtensionsPage.javaScriptURLInput.fill(
+		'https://www.example.com/script.js'
+	);
+
+	await editJSClientExtensionsPage.publish(WaitAction.SUCCESS);
+
+	const clientExtensionUrl =
+		'/o/frontend-data-set-taglib/app/data-set/com_liferay_client_extension_web_internal_portlet_ClientExtensionAdminPortlet-clientExtensionTypes/com_liferay_client_extension_web_internal_portlet_ClientExtensionAdminPortlet-clientExtensionTypes';
+
+	let clientExtensionTypeUrl = '';
+
+	// Listen for all network requests
+
+	page.on('request', (request) => {
+		const url = request.url();
+
+		if (url.includes(clientExtensionUrl)) {
+			const resourceType = request.resourceType();
+			if (resourceType === 'fetch' || resourceType === 'xhr') {
+				clientExtensionTypeUrl = url;
+			}
+		}
+	});
+
+	await clientExtensionsPage.goto();
+
+	// creating new user
```

Comment:
Can we use some existing rest api to do this? I'd say https://github.com/liferay/liferay-portal/blob/8330054eb4df4ba65f739f1ed06d6d3bc795bef3/modules/test/playwright/pages/users-admin-web/UsersAndOrganizationsPage.ts#L660